### PR TITLE
Fix #6851: Move IPFS settings under Web3 settings

### LINKED ISSUE #6851: Wallet: move IPFS settings to Web3
Currently we have an open [PR](https://github.com/brave/brave-ios/pull/6757) for adding IPFS settings. and an open [PR](https://github.com/brave/brave-ios/pull/6844) for renaming `Wallet` settings to `Web3` settings and adding SNS Domain settings.
After these two PRs get merged in, we will need to move IPFS settings under `Web3` as well and become its own IPFS section.

### PR DESCRIPTION
**Pull request by nuo-xu**
## Summary of Changes
Move IPFS settings code from Client to Wallet 
Rewrote IPFS settings in SwiftUI 
Move IPFS settings UI under Web3 

This pull request fixes #6851

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [ ] Light & dark mode
  - [ ] Different size classes (iPhone, landscape, iPad)
  - [ ] Different dynamic type sizes

## Test Plan:
1. check there is no IPFS under Settings 
2. check there is an IPFS section under Web3
3. check if one can set a custom NFT gateway url 
    - ex. `https://nftstorage.link` /  `https://dweb.link`

## Screenshots:

simulator_screenshot_56923D2F-DF7C-4B91-9F45-2004F81BF291 | simulator_screenshot_1F28BF9B-EA0F-4787-88D7-824507583614 | simulator_screenshot_2BBB266C-F1C6-4EBC-B758-9F7873FF7DC1
--|--|--

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/BraveWallet/Settings/IPFSCustomGatewayView.swift [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  @State private var presentWrongGatewayAlert: Bool = false
+  
+  init(ipfsAPI: IpfsAPI) {
+    self.ipfsAPI = ipfsAPI
+  }
+  
+  var body: some View {
+    List {
+      Section(footer: Text(Strings.Wallet.nftGatewayLongDescription)) {
+        TextField(ipfsAPI.nftIpfsGateway?.absoluteString ?? "", text: $url)
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Settings/IPFSCustomGatewayView.swift]: I think this should unwrap to `"https://"` (the default text when cleared)?

File: Sources/BraveWallet/Settings/IPFSCustomGatewayView.swift [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  private enum SetButtonStatus {
+    case enabled
+    case disabled
+    case loading
+  }
+  
+  private let ipfsAPI: IpfsAPI
+  @State private var url: String = "https://"
+  @State private var setButtonStatus: SetButtonStatus = .disabled
+  @State private var presentWrongGatewayAlert: Bool = false
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Settings/IPFSCustomGatewayView.swift]: nit: `isShowingWrongGatewayAlert` / `isPresentingWrongGatewayAlert`

File: Sources/BraveWallet/Settings/IPFSCustomGatewayView.swift [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      presentWrongGatewayAlert = true
+      return
+    }
+    var testURL = enteredURL
+    testURL.append(pathComponents: IPFSCustomGatewayView.ipfsTestPath, IPFSCustomGatewayView.ipfsTestCID)
+    
+    setButtonStatus = .loading
+    resignFirstResponder()
+    
+    URLSession.shared.dataTask(with: testURL) { data, response, error in
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Settings/IPFSCustomGatewayView.swift]: Can we use async/await apis here?

File: Sources/BraveWallet/Settings/IPFSCustomGatewayView.swift (hunk 1 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Settings/IPFSCustomGatewayView.swift]: fixed in [fa59bc5](https://github.com/brave/brave-ios/pull/6880/commits/fa59bc5e8fdf66b6faffc6228672cfa2b695c87a)

File: Sources/BraveWallet/Settings/IPFSCustomGatewayView.swift (hunk 2 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Settings/IPFSCustomGatewayView.swift]: fixed in [fa59bc5](https://github.com/brave/brave-ios/pull/6880/commits/fa59bc5e8fdf66b6faffc6228672cfa2b695c87a)

File: Sources/BraveWallet/Settings/IPFSCustomGatewayView.swift (hunk 3 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Settings/IPFSCustomGatewayView.swift]: fixed in [fa59bc5](https://github.com/brave/brave-ios/pull/6880/commits/fa59bc5e8fdf66b6faffc6228672cfa2b695c87a)

**Review by StephenHeaps — APPROVED** [review]
Missed on last Web3 settings PR `Reset Brave Wallet` row is missing row background colour (only noticeable in dark mode), and noticed `DappsSettings` is as well. Can we fix these here before merge?
Screenshot 2023-02-06 at 1 48 41 PM

**Comment by nuo-xu** [comment]
updated in [83e1a22](https://github.com/brave/brave-ios/pull/6880/commits/83e1a22f922b9a970c73465e5950676ee87f5b22)

**Pull request merged**